libraries/platforms/dpi-verilator/bsg_manycore_platform.cpp

```diff
@@ -545,3 +547,19 @@ int hb_mc_platform_log_disable(hb_mc_manycore_t *mc){
         hb_mc_platform_t *pl = reinterpret_cast<hb_mc_platform_t *>(mc->platform);
         return hb_mc_tracer_log_disable(pl->tracer);
 }
+
+/**
+ * Check if chip reset has completed.
+ * @param[in] mc    A manycore instance initialized with hb_mc_manycore_init()
+ * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+ */        
+int hb_mc_platform_reset_is_done(hb_mc_manycore_t *mc)
+{
+        hb_mc_platform_t *pl = reinterpret_cast<hb_mc_platform_t*>(mc->platform);
+        manycore_pr_dbg(mc, "%s: calling eval()\n", __func__);
+        pl->top->eval();
+        bool done;
```

Comment:
I'm not sure this is the right place for this eval call. Did you remove the reset done check from tx? There was a while loop in init, right?

Reply:
I would rather call this reset_is_done, and make a reset_done_wait function.

Reply:
I did remove the reset done check from tx. It's now in init.